crates/kubelet/src/kubelet.rs

```diff
@@ -44,6 +44,11 @@ impl<P: Provider> Kubelet<P> {
         kube_config: kube::Config,
         config: Config,
     ) -> anyhow::Result<Self> {
+
+        if config.server_config.private_key_file == config.server_config.cert_file {
+            panic!("The config private key file and server config certificate file are the same");
```

Comment:
`panic` probably isn't the right choice here... That will cause the program to exit abruptly. `panic` should only be used when a program reaches an unrecoverable state.

It might be better to instead use the `anyhow::Error` return type. Something like this...

```
return Err(anyhow!("The config private key file and server config certificate file are the same"));
```